agentMET4FOF/agents.py

```diff
@@ -74,9 +71,16 @@ def step(self):
         self.agent_loop()
 
     def _remove_methods(self, cls):
+        """Remove methods from the other backends base class from the current agent"""
         for name in list(vars(cls)):
             if not name.startswith("__"):
-                delattr(cls, name)
+                try:
+                    delattr(self, name)
```

Comment:
The important bit here is the change from `cls` to `self` which results in removing the methods from the current object, not from the class.